File: src/test/java/io/github/dmitrikudrenko/calculator/CalculatorAbstractTest.java
package io.github.dmitrikudrenko.calculator;

import io.appium.java_client.AppiumDriver;
import io.github.dmitrikudrenko.core.AbstractTest;
import io.github.dmitrikudrenko.core.AppiumDriverBuilder;

import java.io.IOException;

public abstract class CalculatorAbstractTest extends AbstractTest<CalculatorApplication> {
    protected CalculatorScreen calculatorScreen;

    @Override
    protected void onStart(CalculatorApplication application) {
        calculatorScreen = new CalculatorScreen(application.getDriver());
    }

    @Override
    protected CalculatorApplication create(AppiumDriver driver) {
        return new CalculatorApplication(driver);
    }

    @Override
    protected AppiumDriverBuilder.AndroidDriverBuilder createAndroidDriverBuilder() throws IOException {
        return super.createAndroidDriverBuilder()
                .withPackageName(CalculatorConstants.PACKAGE_NAME)
                .withLaunchActivity(CalculatorConstants.LAUNCH_ACTIVITY);
    }
}

File: src/test/kotlin/io/github/dmitrikudrenko/calculator/core/CalculatorAbstractTest.kt
package io.github.dmitrikudrenko.calculator.core

import io.appium.java_client.AppiumDriver
import io.github.dmitrikudrenko.core.AbstractTest
import io.github.dmitrikudrenko.core.AppiumDriverBuilder

import java.io.IOException

abstract class CalculatorAbstractTest : AbstractTest<CalculatorApplication>() {
    protected lateinit var calculatorScreen: CalculatorScreen

    override fun onStart(application: CalculatorApplication) {
        calculatorScreen = application.calculatorScreen()
    }

    override fun create(driver: AppiumDriver<*>): CalculatorApplication {
        return CalculatorApplication(driver)
    }

    @Throws(IOException::class)
    override fun createAndroidDriverBuilder(): AppiumDriverBuilder.AndroidDriverBuilder {
        return super.createAndroidDriverBuilder()
                .withPackageName(CalculatorConstants.PACKAGE_NAME)
                .withLaunchActivity(CalculatorConstants.LAUNCH_ACTIVITY)
    }
}
